### PR TITLE
[FileState] Deprecation enactment for 3.0.0

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

``modname`` and ``msg_store`` are now required to be given in FileState, ``collect_block_lines`` has also been removed.

### CODE REVIEW

File: pylint/lint/parallel.py [hunk 1]
```diff
@@ -165,6 +165,7 @@ def check_parallel(
             msg_status,
             mapreduce_data,
         ) in executor.map(_worker_check_single_file, files):
+            assert isinstance(base_name, str)
```
Comment (DanielNoord) [inline pylint/lint/parallel.py:168]: This should now always be `str` without an assertion

Comment (Pierre-Sassoulas) [reply pylint/lint/parallel.py:168]: mypy does not like it:
```
pylint/lint/parallel.py:168: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")  [assignment]

Found 1 error in 1 file (checked 3 source files)
```

Comment (DanielNoord) [reply pylint/lint/parallel.py:168]: The typing on `PyLinter` should be updated!

Comment (Pierre-Sassoulas) [reply pylint/lint/parallel.py]: I rebased so we can rebase and merge, there was quite a lot more to change than I thought :D

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**